src/reflect/type.go

```diff
@@ -995,6 +1001,10 @@ func (t *rawType) AssignableTo(u Type) bool {
 		return true
 	}
 
+	if t.underlying() == u.(*rawType).underlying() && (!t.isNamed() || !u.(*rawType).isNamed()) {
```

Comment:
See https://go.dev/ref/spec#Assignability

If the underlying types are the same, only one can be named.